org.eclipse.scanning.sequencer/src/org/eclipse/scanning/sequencer/nexus/SolsticeScanMonitor.java

```diff
@@ -141,6 +155,24 @@ public NXcollection createNexusObject(NexusScanInfo info) {
 		scanFinished.setFillValue(0);
 		scanPointsCollection.createDataNode(FIELD_NAME_SCAN_FINISHED, scanFinished);
 		
+		// write the scan shape
+		int[] shape = info.getShape();
+		logger.info("Estimated scan shape " + Arrays.toString(shape));
+		scanPointsCollection.setDataset(FIELD_NAME_SCAN_SHAPE, DatasetFactory.createFromObject(shape));
+		
+		// write the estimated scan duration
+		long estimatedScanTimeMillis = model.getScanInformation().getEstimatedScanTime();
+		String estimateScanTimeStr = durationInMillisToString(Duration.ofMillis(estimatedScanTimeMillis));
+		logger.info("Estimated scan time " + estimateScanTimeStr);
+		scanPointsCollection.setField(FIELD_NAME_SCAN_ESTIMATED_DURATION, estimateScanTimeStr);
+		
+		scanDuration = new LazyWriteableDataset(FIELD_NAME_SCAN_DURATION, String.class,
+				new int[] { 1 }, new int[] { -1 }, new int[] { 1 }, null);
+		scanPointsCollection.createDataNode(FIELD_NAME_SCAN_DURATION, scanDuration);
+		scanStartTime = Instant.now(); // record the current time
+		// TODO: should use @ScanStart but add this monitor to the scanmodel after the annotation manager is created
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1135)